src/profile-logic/import/dhat.js

```diff
@@ -0,0 +1,324 @@
+  // - All modes: A mandatory array of integers.
+  fs: IndexIntoDhatFrames[],
+|};
+
+// All units of time are in instruction counts.
+// Per: https://valgrind.org/docs/manual/dh-manual.html
+//   As with the Massif heap profiler, DHAT measures program progress by counting
+//   instructions, and so presents all age/time related figures as instruction counts.
+//   This sounds a little odd at first, but it makes runs repeatable in a way which
+//   is not possible if CPU time is used.
+type InstructionCounts = number;
+type Blocks = number;
+type IndexIntoDhatFrames = number;
+type ReadCount = number;
+type WriteCount = number;
+
+export function attemptToConvertDhat(json: mixed): Profile | null {
+  if (!json || typeof json !== 'object') {
+    return null;
+  }
+
+  const { dhatFileVersion } = json;
+  if (typeof dhatFileVersion !== 'number') {
+    // This is not a dhat file.
+    return null;
+  }
+
+  if (dhatFileVersion !== 2) {
+    throw new Error(
+      `This importer only supports dhat version 2. The file provided was version ${dhatFileVersion}.`
+    );
+  }
+  const dhat = coerce<mixed, DhatJson>(json);
+
+  const profile = getEmptyProfile();
+  profile.meta.product = dhat.cmd + ' (dhat)';
+  profile.meta.importedFrom = `dhat`;
+
+  const allocations = getEmptyUnbalancedNativeAllocationsTable();
+  const thread = getEmptyThread();
+  thread.pid = dhat.pid;
+  const { funcTable, stringTable, stackTable, frameTable } = thread;
+
+  const funcKeyToFuncIndex = new Map<string, IndexIntoFuncTable>();
+
+  // dhat profiles do no support categories. Fill the category and subcategory information
+  // with 0s.
+  const otherCategory = 0;
+  const otherSubCategory = 0;
+
+  // Convert the frame table.
+  for (let funcName of dhat.ftbl) {
+    let fileName = dhat.cmd;
+    let address = -1;
+    let line = null;
+    let column = null;
+
+    const result = funcName.match(/^(0x[0-9a-f]+): (.+) \((.+):(\d+):(\d+)\)$/);
+    // ^(0x[0-9a-f]+): (.+) \((.+):(\d+):(\d+)\)$   Regex
+    //  (1          )  (2 )   (3 ) (4  ) (5  )      Capture groups
+    // ^                                        $   Start to end
+    //               :      \(                \)    Some raw characters
+    //  (0x[0-9a-f]+)                               Match the address, e.g. 0x10250148c
+    //                 (.+)                         Match the function name
+    //                        (.+)                  Match the filename
+    //                             (\d+)            Match the line number
+    //                                   (\d+)      Match the column number
+
+    // Example input: "0x10250148c: alloc::vec::Vec<T,A>::append_elements (vec.rs:1469:9)"
+    // Capture groups: 11111111111  2222222222222222222222222222222222222  333333 4444 5
+    if (result) {
+      address = parseInt(result[1].substr(2), 16);
+      funcName = result[2];
+      fileName = result[3];
+      line = Number(result[4]);
+      column = Number(result[5]);
+    }
```

Comment:
```
   // If the above regex doesn't match, just use the raw funcName, without additional
    // information.
```

I'm adding this comment, I think it would be useful to still have the raw funcName, even without the additional information.

Reply:
Got it, thanks